pkg/utils/utils.go

```diff
@@ -169,6 +177,24 @@ func configSriovDevice(iface *sriovnetworkv1.Interface, ifaceStatus *sriovnetwor
 		if err != nil {
 			glog.Warningf("configSriovDevice(): unable to parse VFs for device %+v %q", iface.PciAddress, err)
 		}
+		// Set Sriov mode for ETH devices
+		if ifaceStatus.Name != "" {
+			if link, err := netlink.LinkByName(ifaceStatus.Name); err == nil && link.Attrs().EncapType == "ether" {
+				for _, vf := range vfs {
+					if err := Unbind(vf); err != nil {
+						glog.Warningf("configSriovDevice(): fail to unbind driver for device %s", vf)
+					}
+				}
+				if err := setNicSriovMode(iface.PciAddress, iface.SriovMode); err != nil {
```

Comment:
If the hardware doesn't support "switchdev" mode, the `iface.SriovMode` would be empty string, right? If that is the case, I think it would be better to check if `iface.SriovMode` is empty before invoking `setNicSriovMode`.

Reply:
I agree